### PR TITLE
Fix negative pixels that could happen from bleed trails

### PR DESCRIPTION
**Pull request by rmjarvis**
Fix a bug in the bleed trail algorithm that would subtract too much from excess charge when dealing with off-the-bottom pixels and less than a full well of charge was left to deal with.

### CODE REVIEW

**Review by jchiang87 — APPROVED** [review]
Thanks!  LGTM.

File: imsim/lsst_image.py [hunk 1]
```diff
@@ -293,11 +294,15 @@ def addNoise(self, image, config, base, image_num, obj_num, current_var, logger)
             camera = get_camera(self.camera_name)
             det_name = base['det_name']
             serial_number = camera[det_name].getSerial()
+            # Note: the regular Python hash function is non-derterministic, which is not good.
```
Comment (erykoff) [inline imsim/lsst_image.py:297]: non-derterministic -> non-deterministic

Comment (rmjarvis) [reply imsim/lsst_image.py:297]:
```suggestion
            # Note: the regular Python hash function is non-deterministic, which is not good.
```

**Comment by jchiang87** [comment]
The most recent set of changes look fine to me.  Unless I hear otherwise, I'll go ahead and merge this at 5pm PT today.

**Pull request merged**